### PR TITLE
[Gohai][ASC-663] Implement a typed API for the processes package, add unit tests

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Implement a simple typed API for the `processes` package, and implement the old untyped API on top of it.
Add unit tests to verify that the data is coherent and has the expected format.

### Motivation

Having a typed API for the `processes` package and testing it properly (there were no tests before). 

### Additional Notes

The first commit renames two of the files, so it should probably be reviewed separately from the rest.

The package output is very legacy, it is an array with different types, meaning it can't be easily unmarshalled and checked.

### Possible Drawbacks / Trade-offs

There should be no functional change for the old API.

### Describe how to test/QA your changes

Check that there is no change in the generated json. It should be *exactly* the same format (note that the information will change every time you compute them since it's about running processes).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/gohai/processes/processes.go [hunk 1]
```diff
@@ -3,27 +3,65 @@
...
+}
+
+// Processes is the Collector type of the processes package.
+type Processes struct{}
+
+const name = "processes"
+
+func init() {
+	flag.IntVar(&options.limit, name+"-limit", 20, "Number of process groups to return")
+}
```
Comment (hush-hush) [inline pkg/gohai/processes/processes.go:26]: Do  we still need this ? Same for `var options struct {`.

File: pkg/gohai/processes/processes.go [hunk 2]
```diff
@@ -3,27 +3,65 @@
...
 type ProcessField [7]interface{}
 
-// getProcesses return a JSON payload which is compatible with
-// the legacy "processes" resource check
-func getProcesses(limit int) ([]interface{}, error) {
-	processGroups, err := gops.TopRSSProcessGroups(limit)
+// Collect collects the processes information.
+// Returns an object which can be converted to a JSON or an error if nothing could be collected.
+// Tries to collect as much information as possible.
+func (processes *Processes) Collect() (result interface{}, err error) {
```
Comment (hush-hush) [inline pkg/gohai/processes/processes.go:63]: we don't use the returned variable names anywhere.

```suggestion
func (processes *Processes) Collect() (interface{}, error) {
```

File: pkg/gohai/processes/processes_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+func TestGet(t *testing.T) {
+	processes, err := Get()
+	require.NoError(t, err)
+
+	require.NotEmpty(t, processes)
+	for _, process := range processes {
+		assert.NotEmpty(t, process.Name)
+
+		assert.NotEmpty(t, process.Usernames)
+		assert.IsIncreasing(t, process.Usernames)
```
Comment (hush-hush) [inline pkg/gohai/processes/processes_test.go:27]: Do we guarantee that usernames are sorted in `/pkg/gohai/processes/gops`  ?

**Review by hush-hush — APPROVED** [review]
Looks good ! I added a question and a nit-pick

File: pkg/gohai/processes/processes_test.go (hunk 3 above)
Comment (pgimalac) [reply pkg/gohai/processes/processes_test.go:27]: Yes, cf https://github.com/DataDog/datadog-agent/blob/main/pkg/gohai/processes/gops/process_group.go#L54.
It wasn't tested before so I added it to the test.

File: pkg/gohai/processes/processes.go (hunk 2 above)
Comment (pgimalac) [reply pkg/gohai/processes/processes.go:63]: Nice catch !

File: pkg/gohai/processes/processes_test.go (hunk 3 above)
Comment (hush-hush) [reply pkg/gohai/processes/processes_test.go:27]: Should this test be in `pkg/gohai/processes/gops` then ?

Comment (pgimalac) [reply pkg/gohai/processes/processes_test.go:27]: I guess it depends on whether that guarantee is for `gops` specifically or the full processes package, ie. "Should users of the package assume that the usernames are sorted ?"
As I specified in the doc of the `ProcessGroup` type (which is the exported type of the package) that the usernames are sorted, I think we can keep that at the package level

**Pull request merged**